app/routes/chatbot.js

```diff
@@ -300,89 +332,131 @@ router.use((req, res, next) => {
-        return false;
+      if (req.timedout) {
+        return helpers.sendTimeoutResponse(res);
       }
+      req.signup = signup; // eslint-disable-line no-param-reassign
 
-      if (scope.broadcast_id) {
-        // TODO: Add new parameter for broadcast_id to Signup post instead of saving here.
-        scope.signup.broadcast_id = scope.broadcast_id;
-        scope.signup.save().catch((err) => logger.error('Error saving broadcast id', err.message));
-      }
+      return next();
+    })
+    .catch(err => helpers.sendErrorResponse(res, err));
+});
 
-      if (isCommand(scope.incoming_message, 'member_support')) {
-        scope.cmd_member_support = true;
-        scope.oip = agentViewOip;
-        return 'member_support';
-      }
+/**
+ * Sanity check: make sure there's a Signup.
+ */
+router.use((req, res, next) => {
+  if (!req.signup) {
+    return helpers.sendResponse(res, 500, 'req.signup is undefined');
+  }
 
-      if (scope.signup.draft_reportback_submission) {
-        logger.debug(`draft_reportback_submission:${scope.signup.draft_reportback_submission._id}`);
-        return controller.continueReportbackSubmission(scope);
-      }
+  logger.verbose(`user:${req.user._id} campaign:${req.campaign.id} signup:${req.signup._id}`);
+  newrelic.addCustomParameters({ signupId: req.signup._id });
 
-      if (isCommand(scope.incoming_message, 'reportback')) {
-        return scope.signup.createDraftReportbackSubmission().then(() => 'ask_quantity');
-      }
+  return next();
+});
 
-      if (scope.signup.reportback) {
-        if (scope.keyword || scope.broadcast_id) {
-          return 'menu_completed';
-        }
-        // If we're this far, member didn't text back Reportback or Member Support commands.
-        return 'invalid_cmd_completed';
-      }
+/**
+ * Check for non-reportback conversation messages first for sending reply message.
+ */
+router.post('/', (req, res, next) => {
+  if (isCommand(req.incoming_message, 'member_support')) {
+    return endConversationWithMessageType(req, res, 'member_support');
+  }
 
-      if (scope.keyword || scope.broadcast_id) {
-        return 'menu_signedup_gambit';
-      }
+  // If this is a reportback conversation, skip to our next middleware.
+  if (req.signup.draft_reportback_submission || isCommand(req.incoming_message, 'reportback')) {
+    return next();
+  }
 
-      return 'invalid_cmd_signedup';
-    })
-    .then((msgType) => {
-      // This is hacky, CampaignBotController.postReportback returns error that isn't caught.
-      // TODO: Clean this up when ready to take on https://github.com/DoSomething/gambit/issues/744.
-      if (msgType instanceof Error) {
-        throw new Error(msgType.message);
-      }
+  // If member has completed this campaign:
+  if (req.signup.reportback) {
+    // And it's the beginning of a conversation:
+    if (req.keyword || req.broadcast_id) {
+      return continueConversationWithMessageType(req, res, 'menu_completed');
```

Comment:
Doe. `continueConversationWithMessageType` doesn't have the check for a `{{quantity}}` replace yet (it's getting set later) -- so this introduces a bug for the Completed Menu:
![image uploaded from ios 3](https://cloud.githubusercontent.com/assets/1236811/25050797/4d6ad998-20fe-11e7-9674-2659bc0466b9.jpg)

Fix comin' right up 🍳 🍴